smartsim/entity/dbobject.py

```diff
@@ -130,16 +131,16 @@ def _enumerate_devices(self) -> t.List[str]:
 
     @staticmethod
     def _check_devices(
-        device: t.Literal["CPU", "GPU"],
+        device: str,
```

Comment:
If `_check_devices` accepts a `Device` enum, the check `if ":" in device:` will always be false right? And when I cast the device to an enum, I could chop off the `:#` part if exists but isnt this necessary for the func?

Reply:
Oh! Is that valid input?? Yikes, then that `t.Literal["CPU", "GPU"]` type hint was not correct 😅 

Uhhh, give me another chance to re-look through that, but yeah, that may unfortunately need to stay a string.

Reply:
I talked with @al-rigazzi about this, it looks like the `"{DEVICE}:{NUM}"` style input is relic from the days before the `first_device` parameter was introduced. This is leading to some ambiguities in the API.

This definitely should be addressed, but if it is going to cause an API break I'd rather keep it out of this PR, just in case we need to revert! I will post a ticket to address, but for now this implementation looks good to me!!

Reply:
Issue regarding the ambiguous `DBObject` API has been made at #534

Reply:
thanks for looking into that and creating the ticket!